Specs gtfs2ntfs: reading calendars & common pre-processing (#438)
CanalTP/transit_model: src/documentation/common.md

# Shared specifications for all converters
This document explains the shared parts among all the converters when converting a 

> data set from a given format into a [NTFS](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md#physical_modestxt-requis) dataset.

Reviewer: ```suggestion
data set from a given format into a [NTFS](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md) dataset.
```

## Data prefix
The construction of NTFS objects IDs requires, for uniqueness purpose, that a unique 

> prefix (specified for each source of data as an addtional parameter to each converter)

Reviewer: ```suggestion
prefix (specified for each source of data as an additional parameter to each converter)
```

needs to be included in every object's id.

Prepending all the identifiers with a unique prefix ensures that the NTFS identifiers are unique accross all the NTFS datasets. With this assumption, merging two NTFS datasets can be done without worrying about conflicting identifiers.

> This prefix should be applied to all NTFS identifier except for the physical_mode identifiers that are standarized and fixed values. Fixed values are described in the [NTFS specifications](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md#physical_modestxt-requis)

Reviewer: you can write `physical mode` instead of `physical_mode`

and except for commercial mode.

## Configuration of each converter
A configuration file `config.json`, as it is shown below, is provided for each 
converter and contains additional information about the data source as well as about 
the upstream system that generated the data (if available). In particular, it 

> provides the necessary information for the required NTFS files datasets.txt and contributors.txt.

Reviewer: I would rather write "datasets.txt and contributors.txt" like
`datasets.txt` and `contributors.txt`
or **datasets.txt** and **contributors.txt**
or whatever else

Reviewer 2: ```suggestion
provides the necessary information for the required NTFS files datasets.txt and contributors.txt. Some additional metadata can also be inserted in the file `feed_infos.txt`.
```

```json
{
    "contributor": {
        "contributor_id": "DefaultContributorId",
        "contributor_name": "DefaultContributorName",
        "contributor_license": "DefaultDatasourceLicense",
        "contributor_website": "http://www.default-datasource-website.com"
    },
    "dataset": {
        "dataset_id": "DefaultDatasetId"
    },
    "feed_infos": {
        "feed_publisher_name": "DefaultContributorName",
        "feed_license": "DefaultDatasourceLicense",
        "feed_license_url": "http://www.default-datasource-website.com",
    }
}
```
The objects `contributor` and `dataset` are required, containing at least the 
corresponding identifier (and the name for `contributor`), otherwise the conversion 
stops with an error. The object `feed_infos` is optional.

> The files datasets.txt and contributors.txt provide additional information about the data source.

Reviewer: same here

In general I think we should write files in bold or something else

### Loading Dataset

| NTFS file | NTFS field | key in `config.json` | Constraint | Note |
| --- | --- | --- | --- | ---
| datasets.txt | dataset_id | dataset_id | Required | This field is prefixed.
| datasets.txt | contributor_id | contributor_id | Required | This field is prefixed.

> | datasets.txt | dataset_start_date |  |  | Date of the first trip of the dataset (including all lines of all networks).

Reviewer 2: This formulation feels wrong to me because it's not about a "date of the dataset" but about a "date of a trip", isn't it?

> | datasets.txt | dataset_start_date |  |  | Date of the first trip of the dataset (including all lines of all networks).

Reviewer 2: You're right too, indeed :-D Then maybe something like this?
```suggestion
| datasets.txt | dataset_start_date |  |  | First date of the first trip of the dataset (including all lines of all networks).
```

| datasets.txt | dataset_end_date |  |  | Date of the last trip of the dataset (including all lines of all networks).

> ### Loading Contributor

Reviewer 2: It would feel logical to have the section about the contributor before the section about the dataset (since a dataset refers to a contributor).

# Shared specifications for all converters
This document explains the shared parts among all the converters when converting a 
data set from a given format into a [NTFS](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md) dataset.

## Data prefix
The construction of NTFS objects IDs requires, for uniqueness purpose, that a unique 
prefix (specified for each source of data as an additional parameter to each converter)
needs to be included in every object's id.

Prepending all the identifiers with a unique prefix ensures that the NTFS identifiers are unique accross all the NTFS datasets. With this assumption, merging two NTFS datasets can be done without worrying about conflicting identifiers.

> This prefix should be applied to all NTFS identifier except for the physical mode identifiers that are standarized and fixed values. Fixed values are described in the [NTFS specifications](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md#physical_modestxt-requis)

Reviewer 2: ```suggestion
This prefix should be applied to all NTFS identifiers except for the physical mode identifiers that are standardized and fixed values. Fixed values are described in the [NTFS specifications](https://github.com/CanalTP/ntfs-specification/blob/master/ntfs_fr.md#physical_modestxt-requis)
```
